pkg/auth/service.go

```diff
@@ -959,3 +964,626 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
+	resp, err := a.apiClient.CreateUserWithResponse(ctx, CreateUserJSONRequestBody{
+		Username: user.Username,
+	})
+	if err != nil {
+		return InvalidUserID, err
+	}
+	if err := expectedStatusCode(resp, 201); err != nil {
+		return InvalidUserID, err
+	}
+
+	return resp.JSON201.Id, nil
+}
+
+func (a *APIAuthService) DeleteUser(ctx context.Context, username string) error {
+	u, err := a.GetUser(ctx, username)
+	if err != nil {
+		return err
+	}
+	resp, err := a.apiClient.DeleteUserWithResponse(ctx, u.ID)
+	if err != nil {
+		return err
+	}
+	return expectedStatusCode(resp, 204)
+}
+
+func (a *APIAuthService) GetUserByID(ctx context.Context, userID int) (*model.User, error) {
+	return a.cache.GetUserByID(userID, func() (*model.User, error) {
+		resp, err := a.apiClient.GetUserByIdWithResponse(ctx, userID)
+		if err != nil {
+			return nil, err
+		}
+		// TODO(Guys): consider  support 404 in this case
+
+		if err := expectedStatusCode(resp, 200); err != nil {
+			return nil, err
+		}
+		u := resp.JSON200
+		return &model.User{
+			ID:                u.Id,
+			CreatedAt:         time.Unix(u.CreationDate, 0),
+			Username:          u.Name,
+			FriendlyName:      u.FriendlyName,
+			Email:             u.Email,
+			EncryptedPassword: nil,
+			Source:            "",
+		}, nil
+	})
+}
+
+func (a *APIAuthService) GetUser(ctx context.Context, username string) (*model.User, error) {
+	return a.cache.GetUser(username, func() (*model.User, error) {
+		resp, err := a.apiClient.GetUserByNameWithResponse(ctx, &GetUserByNameParams{UserName: username})
+		if err != nil {
+			return nil, err
+		}
+		if err := expectedStatusCode(resp, 200); err != nil {
+			return nil, err
+		}
+		u := resp.JSON200
+		return &model.User{
+			ID:                u.Id,
+			CreatedAt:         time.Unix(u.CreationDate, 0),
+			Username:          u.Name,
+			FriendlyName:      u.FriendlyName,
+			Email:             u.Email,
+			EncryptedPassword: nil,
+			Source:            "",
+		}, nil
+	})
+}
+
+func (a *APIAuthService) GetUserByEmail(ctx context.Context, email string) (*model.User, error) {
+	return a.cache.GetUser(email, func() (*model.User, error) {
+		resp, err := a.apiClient.GetUserByEmailWithResponse(ctx, &GetUserByEmailParams{Email: email})
+		if err != nil {
+			return nil, err
+		}
+		if err := expectedStatusCode(resp, 200); err != nil {
+			return nil, err
+		}
+		u := resp.JSON200
+		user := &model.User{
+			ID:           u.Id,
+			CreatedAt:    time.Unix(u.CreationDate, 0),
+			Username:     u.Name,
+			FriendlyName: u.FriendlyName,
+			Email:        u.Email,
+			Source:       getNonRequiredString(u.Source),
+		}
+		if u.EncryptedPassword != nil {
+			user.EncryptedPassword = *u.EncryptedPassword
+		}
+		return user, err
+	})
+}
+
+// getNonRequiredString returns empty string in case of nil
+func getNonRequiredString(s *string) string {
+	if s == nil {
+		return ""
+	}
+	return *s
+}
+
+func toPagination(paginator Pagination) *model.Paginator {
```

Comment:
same as the above function - looks like a set of functions we require to build api client or api user